src/odemis/acq/align/test/keypoint_test.py

```diff
@@ -0,0 +1,226 @@
+
+Copyright © 2017 Guilherme Stiebler, Éric Piel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+import unittest
+import logging
+from odemis.acq.align import keypoint
+import cv2
+from odemis.util.conversion import get_img_transformation_md
+from odemis.util.dataio import open_acquisition
+import numpy
+import cairo
+import math
+from odemis import model
+from numpy.linalg import inv
+from scipy import ndimage
+
+imgs_folder = './src/odemis/acq/align/test/images/'
+
+
+def preprocess(img, invert, flip, crop, gaussian_sigma, eqhis):
+    '''
+    The pre-processing function like the one on the alignment plugin
+    img (DataArray): Input image
+    invert (bool): Invert the brightness levels of the image
+    flip (tuple(bool, bool)): Determine if the image should be flipped on the X and Y axis
+    crop (tuple(t,b,l,r): Crop values in pixels
+    gaussian_sigma (int): Blur intensity
+    eqhis (bool): Determine if an histogram equalization should be executed
+    return: Processed image
+    '''
+    metadata = img.metadata
+
+    flip_x, flip_y = flip
+    # flip on X axis
+    if flip_x:
+        img = img[:, ::-1]
+
+    # flip on Y axis
+    if flip_y:
+        img = img[::-1, :]
+
+    crop_top, crop_bottom, crop_left, crop_right = crop
+    # remove the bar
+    img = img[crop_top:img.shape[0] - crop_bottom, crop_left:img.shape[1] - crop_right]
+
+    # Invert the image brightness
+    if invert:
+        mn = img.min()
+        mx = img.max()
+        img = mx + mn - img
+
+    # equalize histogram
+    if eqhis:
+        if img.dtype == numpy.uint16:
+            img = cv2.convertScaleAbs(img, alpha=(255.0/65535.0))
+        img = cv2.equalizeHist(img)
+
+    # blur the image using a gaussian filter
+    img = ndimage.gaussian_filter(img, sigma=gaussian_sigma)
+
+    # return a new DataArray with the metadata of the original image
+    return  model.DataArray(img, metadata)
+
+
+class TestKeypoint(unittest.TestCase):
+
+    def test_image_pair(self):
+        ''' Testing a pair of images
+        '''
+        # only one image will be used, but this structure helps to test
+        # different images
+        image_pairs = [
+            (
+                ('Slice69_stretched.tif', True, (False, True), (0, 0, 0, 0), 7),
+                ('g_009_cropped.tif', False, (False, False), (0, 0, 0, 0), 5)
+            ),
+            (
+                ('001_CBS_010.tif', False, (False, False), (0, 0, 0, 0), 0),
```

Comment:
Comment out the second and third image pairs, which are not in the repository. It's fine to have just one pair of image in the test cases, as you have other test cases with synthetic images.